### PR TITLE
ISPN-11914 ServerRunMode.FORKED should use by default the server folder

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-11914

The goal of this PR is not to fix all tests that are broken for `ServerRunMode.FORKED`
The goal is allow we run the FORKED mode using the server that was built

A simple test that is working is: `mvn clean install -DskipTests && mvn verify -pl server/tests -Dit.test=RestOperations -Dorg.infinispan.test.server.driver=FORKED`

I have a plan to introduce a new `Jenkinsfile` that will run the tests for all `ServerRunMode`. The job will be running daily.

### CODE REVIEW

**Comment by diegolovison** [comment]
run performance tests please

**Comment by diegolovison** [comment]
run performance tests please

**Comment by diegolovison** [comment]
run performance tests please

**Comment by diegolovison** [comment]
run performance tests please

**Comment by diegolovison** [comment]
run performance tests please

**Review by rhusar — CHANGES_REQUESTED** [review]
Needs a rebase now :-)

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java [hunk 1]
```diff
@@ -26,38 +34,51 @@
...
+      List<Path> serverHomes = new ArrayList<>();
+      Path src = Paths.get(globalServerHome).normalize();
+      for (int i = 0; i < configuration.numServers(); i++) {
+         Path dest = Paths.get(CommonsTestingUtil.tmpDirectory(), UUID.randomUUID().toString());
+         try {
+            Files.createDirectory(dest);
+            Files.walkFileTree(src, new CommonsTestingUtil.CopyFileVisitor(dest, true));
+            Files.walkFileTree(this.getConfDir().toPath(), new CommonsTestingUtil.CopyFileVisitor(getServerConfDir(dest.toString()), true));
+         } catch (IOException e) {
+            throw new UncheckedIOException("Cannot copy the server to temp folder", e);
```
Comment (rhusar) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/core/ForkedInfinispanServerDriver.java:57]: s/folder/directory :-)

**Comment by diegolovison** [comment]
@rhusar please review

**Comment by rhusar** [comment]
@diegolovison Is this the intended design Diego? While this solves some of the log isolation problems, this copies now cca 110 MB of data per test class per cluster member; we have e.g. 10 test classes, each should run against 2 node infinispan server cluster so its more than 2 GB of temporary data to manage. Opened https://issues.redhat.com/browse/ISPN-13109

**Comment by rhusar** [comment]
@diegolovison Also, have you tested this on windows? It seems to have broken windows support again - opened https://issues.redhat.com/browse/ISPN-13107 and will try to debug this and submit a PR for that as well.

**Comment by diegolovison** [comment]
@rhusar the goal of this PR was to use the current server instead of one provided by the user. If you have two servers now and had two servers before the amount of disk space is/was the same.

**Pull request merged**